src/engine/sparqlExpressions/NumericBinaryExpressions.cpp

```diff
@@ -68,9 +66,198 @@ inline auto andLambda = [](TernaryBool a, TernaryBool b) {
   }
   return Id::makeUndefined();
 };
-NARY_EXPRESSION(AndExpression, 2,
-                FV<decltype(andLambda), EffectiveBooleanValueGetter>,
-                SET<SetOfIntervals::Intersection>);
+
+namespace constructPrefilterExpr {
+
+// _____________________________________________________________________________
+template <typename BinaryPrefilterExpr>
+std::optional<std::vector<PrefilterExprVariablePair>> mergeChildrenAndLogicImpl(
+    std::optional<std::vector<PrefilterExprVariablePair>>&& leftChildExprs,
+    std::optional<std::vector<PrefilterExprVariablePair>>&& rightChildExprs) {
+  if (!leftChildExprs.has_value() || !rightChildExprs.has_value()) {
```

Comment:
I think you can linearize thiis code:
```
if (!left && !right) {
  return nullopt;
}
if (left.has_value() != right.has_value()) {
  return the child with the value.
}
CORRECTNESS_CHECK(both have value)
// Do the more complex case.

Reply:
This can be omitted given that the arguments have args of type `std::vector<PrefilterExprVariablePair>` now.